woocommerce/woocommerce
# Update/implement connect to jetpack

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This is still WIP, but I'm opening it now to receive feedback and unblock other Jetpack login page PRs.

Jetpack Authorization is functioning as expected, but I'm not 100% comfortable.

1. Both [Jetpack Connection Rest Auth](https://github.com/woocommerce/woocommerce/pull/38674/files#diff-aa7d46de8dc31309902e10dd4f0695a9346c08e0fb8d43342b40026d6172aa45R67) and [Config](https://github.com/woocommerce/woocommerce/pull/38674/files#diff-aa7d46de8dc31309902e10dd4f0695a9346c08e0fb8d43342b40026d6172aa45R73) are loaded on every page.  It seems like this is required for Jetpack to work, but I would like to find a way to load them conditionally, as we only need Jetpack when the user selects Jetpack during the core profiler. 
2. I'm setting `slug` as `woocommerce` in the [config](https://github.com/woocommerce/woocommerce/pull/38674/files#diff-aa7d46de8dc31309902e10dd4f0695a9346c08e0fb8d43342b40026d6172aa45R78), but I'm not sure how/where this value is used exactly. I could not find the value on WPCOM.

I'll look into these two next week as we learn more about how Jetpack works.

You can find more about Jetpack Connection setup from the following repos:

- https://github.com/Automattic/client-example
- https://github.com/Automattic/woo-jetpack-connection-poc
- https://github.com/Automattic/jetpack-connection-ui
- https://github.com/Automattic/jetpack-connection/blob/trunk/docs/register-site.md


This was part of https://github.com/woocommerce/team-ghidorah/issues/211, but never implemented in the original [PR](https://github.com/woocommerce/woocommerce/pull/38405).



### How to test the changes in this Pull Request:

In order to generate Jetpack auth URL, your local env must be publicly accessible. Either use Jurassic Tube or Ngrok to expose your local env.

1. Install and activate WooCommerce on a brand new site
1. Install the WooCommerce Beta Tester plugin
1. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper
1. Enable the 'core-profiler' feature flag
1. Visit your public URL/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
4. Go through the new core profiler flow.
5. Make sure to select Jetpack on the plugins page.
6. Click Continue
7. You should be redirected to Jetpack Auth page.

If you want to change `from` value for testing purposes, change the value from this [line](https://github.com/woocommerce/woocommerce/pull/38674/files#diff-ccca16b15b2dfeb741b98e5d2695dbf1d789de8a44fb869e426929f62f681863R222)
